### ISSUE TITLE
InceptionV4 model failed

### ISSUE THREAD

**Issue by Spritea** [GeorgeSeif/Semantic-Segmentation-Suite#90]
## Please specify the following information when submitting an issue:
- **What are your command line arguments?:**
```
--dataset
CamVid
--model
RefineNet
--frontend
InceptionV4
```
- **Have you written any custom code?**:
Nothing.
- **What have you done to try and solve this issue?:**
I checked issue #17, but it didn't help in fact.
- **TensorFlow version?**:
1.15.0
### Describe the problem
Hi,George:
Long time no see! Glad to see you have improved this code continuously! I tested the latest Inception V4 model, but failed unluckily. At first, I use Refinet paired with Resnet50 model to train CamVid using the latest code, and everything works well.Then I only changed frontend type to **InceptionV4**, it failed.
It seems like issue #17, but I didn't change the default crop size, which is 512* 512. 
Maybe there are some contradictions between the model building code and downloaded .ckpt file. 
Or we need to change the default crop size when using InceptionV4 model.
### Source code / logs

```
WARNING:tensorflow:Variable inception_v4/Mixed_7d/Branch_2/Conv2d_0e_3x1/BatchNorm/moving_variance missing in checkpoint models/inception_v4.ckpt
WARNING:tensorflow:Variable inception_v4/Mixed_7d/Branch_3/Conv2d_0b_1x1/weights missing in checkpoint models/inception_v4.ckpt
WARNING:tensorflow:Variable inception_v4/Mixed_7d/Branch_3/Conv2d_0b_1x1/BatchNorm/beta missing in checkpoint models/inception_v4.ckpt
WARNING:tensorflow:Variable inception_v4/Mixed_7d/Branch_3/Conv2d_0b_1x1/BatchNorm/moving_mean missing in checkpoint models/inception_v4.ckpt
WARNING:tensorflow:Variable inception_v4/Mixed_7d/Branch_3/Conv2d_0b_1x1/BatchNorm/moving_variance missing in checkpoint models/inception_v4.ckpt
WARNING:tensorflow:No Variables to restore
WARNING:tensorflow:From /home/spl03/code/SAR_contest/05-update-gitcode/train.py:94: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See tf.nn.softmax_cross_entropy_with_logits_v2.

This model has 84199392 trainable parameters
Loading the data ...

***** Begin training *****
Dataset --> CamVid
Model --> RefineNet
Crop Height --> 512
Crop Width --> 512
Num Epochs --> 200
Batch Size --> 4
Num Classes --> 32
Data Augmentation:
	Vertical Flip --> False
	Horizontal Flip --> False
	Brightness Alteration --> None
	Rotation --> None

Traceback (most recent call last):
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1350, in _do_call
    return fn(*args)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1329, in _run_fn
    status, run_metadata)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.InvalidArgumentError: Incompatible shapes: [4,28,28,256] vs. [4,30,30,256]
	 [[Node: Add_7 = Add[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](ResizeBilinear, Conv_18/BiasAdd)]]
	 [[Node: Mean/_15 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_10273_Mean", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/spl03/code/SAR_contest/05-update-gitcode/train.py", line 196, in <module>
    _,current=sess.run([opt,loss],feed_dict={net_input:input_image_batch,net_output:output_image_batch})
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 895, in run
    run_metadata_ptr)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1128, in _run
    feed_dict_tensor, options, run_metadata)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1344, in _do_run
    options, run_metadata)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1363, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Incompatible shapes: [4,28,28,256] vs. [4,30,30,256]
	 [[Node: Add_7 = Add[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](ResizeBilinear, Conv_18/BiasAdd)]]
	 [[Node: Mean/_15 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_10273_Mean", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

Caused by op 'Add_7', defined at:
  File "/home/spl03/code/SAR_contest/05-update-gitcode/train.py", line 92, in <module>
    network, init_fn = model_builder.build_model(model_name=args.model, frontend=args.frontend, net_input=net_input, num_classes=num_classes, crop_width=args.crop_width, crop_height=args.crop_height, is_training=True)
  File "/home/spl03/code/SAR_contest/05-update-gitcode/builders/model_builder.py", line 57, in build_model
    network, init_fn = build_refinenet(net_input, preset_model = model_name, frontend=frontend, num_classes=num_classes, is_training=is_training)
  File "/home/spl03/code/SAR_contest/05-update-gitcode/models/RefineNet.py", line 176, in build_refinenet
    low[1]=RefineBlock(high[1],low[0]) # High input = ResNet 1/16, Low input = Previous 1/16
  File "/home/spl03/code/SAR_contest/05-update-gitcode/models/RefineNet.py", line 138, in RefineBlock
    fuse = MultiResolutionFusion(rcu_high, low_inputs,n_filters=256)
  File "/home/spl03/code/SAR_contest/05-update-gitcode/models/RefineNet.py", line 108, in MultiResolutionFusion
    return tf.add(conv_low_up, conv_high)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/ops/gen_math_ops.py", line 183, in add
    "Add", x=x, y=y, name=name)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3160, in create_op
    op_def=op_def)
  File "/home/spl03/anaconda3/envs/titanv/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1625, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Incompatible shapes: [4,28,28,256] vs. [4,30,30,256]
	 [[Node: Add_7 = Add[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](ResizeBilinear, Conv_18/BiasAdd)]]
	 [[Node: Mean/_15 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_10273_Mean", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]


Process finished with exit code 1
```

**Comment by GeorgeSeif** [comment]
I finally got a GPU :)

Hmm, good find!

I looked into the InceptionV4 code and model and it it looks like it's just the way the model is :( Normally, Inception isn't actually used for segmentation. Basically, the way the downsample you end up with an "odd number" dimension at pool5, so that when you upsample the rounding gets messed up! I saw this because if you try InceptionV4 with DeepLabV3 it works just fine (DeepLabV3 uses only up to pool4).

Basically, it looks like a network limitation unfortunately. I'll look into adding another frontend instead. I've been looked around for a dilated resnet but I haven't found one with the pretrained weights yet.

Thanks for the find,
Cheers!
George

**Comment by smajida** [comment]
Hi @GeorgeSeif !

have you had a chance to check InceptionV4. Finally after a long break, I am back and I just checked those CVPR and ECCV18 models. So far all of the models except DDSC(a new issue opened) run smoothly, but it seems there is a problem with InceptionV4 as frontend. no matter which algorithm is chosen, no weights is copied and the following message is shown:

```WARNING:tensorflow:No Variables to restore
This model has 41605070 trainable parameters
Loading the data ...

***** Begin training *****
Dataset --> SingleVehicleDLR_512
Model --> GCN
Crop Height --> 512
Crop Width --> 512
Num Epochs --> 300
Batch Size --> 1
Num Classes --> 2
Data Augmentation:
	Vertical Flip --> False
	Horizontal Flip --> False
	Brightness Alteration --> None
	Rotation --> None

Traceback (most recent call last):
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1327, in _do_call
    return fn(*args)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1306, in _run_fn
    status, run_metadata)
  File "/usr/lib/python3.5/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/framework/errors_impl.py", line 466, in raise_exception_on_not_ok_status
    pywrap_tensorflow.TF_GetCode(status))
tensorflow.python.framework.errors_impl.InvalidArgumentError: Incompatible shapes: [1,30,30,21] vs. [1,28,28,21]
	 [[Node: Add_4 = Add[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](Add_3, Conv2d_transpose/BiasAdd)]]
	 [[Node: Mean/_15 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_10883_Mean", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train.py", line 197, in <module>
    _,current=sess.run([opt,loss],feed_dict={net_input:input_image_batch,net_output:output_image_batch})
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 895, in run
    run_metadata_ptr)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1124, in _run
    feed_dict_tensor, options, run_metadata)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1321, in _do_run
    options, run_metadata)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1340, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Incompatible shapes: [1,30,30,21] vs. [1,28,28,21]
	 [[Node: Add_4 = Add[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](Add_3, Conv2d_transpose/BiasAdd)]]
	 [[Node: Mean/_15 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_10883_Mean", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

Caused by op 'Add_4', defined at:
  File "train.py", line 95, in <module>
    network, init_fn = model_builder.build_model(model_name=args.model, frontend=args.frontend, net_input=net_input, num_classes=num_classes, crop_width=args.crop_width, crop_height=args.crop_height, is_training=True)
  File "/home/majid/Myprojects/Semantic-Segmentation-Suite/neu_version/builders/model_builder.py", line 74, in build_model
    network, init_fn = build_gcn(net_input, preset_model = model_name, frontend=frontend, num_classes=num_classes, is_training=is_training)
  File "/home/majid/Myprojects/Semantic-Segmentation-Suite/neu_version/models/GCN.py", line 71, in build_gcn
    down_4 = tf.add(down_4, down_5)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/ops/gen_math_ops.py", line 80, in add
    result = _op_def_lib.apply_op("Add", x=x, y=y, name=name)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 2630, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/home/majid/.virtualenvs/cv-py3.5/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1204, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Incompatible shapes: [1,30,30,21] vs. [1,28,28,21]
	 [[Node: Add_4 = Add[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](Add_3, Conv2d_transpose/BiasAdd)]]
	 [[Node: Mean/_15 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_10883_Mean", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]
```
